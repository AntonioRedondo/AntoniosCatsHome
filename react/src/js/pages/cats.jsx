import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import actionCreators from "../redux/actionCreators";

import CardFilter from "../components/cardFilter.jsx";
import CardList from "../components/cardList.jsx";

export function Cats(props) {
	return (
		<div>
			<CardFilter onChange={ props.setFilterString } value={ props.filterString } />
			<CardList/>
		</div>
	);
}

Cats.propTypes = {
	filterString: PropTypes.string,
	setFilterString: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
	filterString: state.filterString
});
const mapDispatchToProps = dispatch => ({
	setFilterString: event => dispatch(actionCreators.setFilterString(event.target.value))
});

export default connect(mapStateToProps, mapDispatchToProps)(Cats);
