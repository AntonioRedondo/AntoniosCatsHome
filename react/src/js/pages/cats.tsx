import React from "react";
import { connect } from "react-redux";

import actionCreators from "../redux/actionCreators";

import CardFilter from "../components/cardFilter";
import CardList from "../components/cardList";

interface ICatsProps {
	filterString?: string,
	setFilterString?: (event: React.ChangeEvent<HTMLInputElement>) => void, // It should be compulsory, but annoying TS error complains if compulsory: semantic error TS2322 Type '{}' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<
}

export function Cats(props: ICatsProps) {
	return (
		<div>
			<CardFilter onChange={ props.setFilterString } value={ props.filterString } />
			<CardList/>
		</div>
	);
}

const mapStateToProps = state => ({
	filterString: state.filterString
});
const mapDispatchToProps = dispatch => ({
	setFilterString: event => dispatch(actionCreators.setFilterString(event.target.value))
});

export default connect<{}, {}, ICatsProps>(mapStateToProps, mapDispatchToProps)(Cats);
