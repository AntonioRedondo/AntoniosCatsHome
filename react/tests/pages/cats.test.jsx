import React from "react";
import ReactShallowRenderer  from "react-test-renderer/shallow";
import "jest-styled-components";

import { Cats } from "../../src/js/pages/cats.jsx";


describe("Given the CardList component", () => {
	const shallowRenderer = new ReactShallowRenderer ();
	let tree;
	let mockRequiredPropFunction = () => {};

	beforeEach(() => {
		shallowRenderer.render(
			<Cats
				setFilterString={ mockRequiredPropFunction } />
		);
		tree = shallowRenderer.getRenderOutput();
	});
	
	describe("When the component is mounted", () => {
		it("Should render children", () => {
			expect(tree).toMatchSnapshot();
		});
	});
	
});
