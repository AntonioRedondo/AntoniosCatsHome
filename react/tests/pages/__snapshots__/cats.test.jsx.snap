// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CardList component When the component is mounted Should render children 1`] = `
<div>
  <CardFilter
    onChange={[Function]}
    value={undefined}
  />
  <Connect(CardList) />
</div>
`;
